tests/test_swift.py

```diff
@@ -29,5 +78,100 @@ def test_frames(self):
         self.assertIsInstance(self.app.frames(), Iterable)
 
 
+class SwiftTest(unittest.TestCase):
+    """Unit test for Swift class in swift.py"""
+
+    def setUp(self):
+        """Create a QApplication and a Swift object every time."""
+        self.qapp = QApplication(sys.argv)
+        importlib.import_module = MagicMock()
+        for appInfo in APP_INFOS.values():
+            app = MagicMock()
+            app.frames.return_value = (QWidget(),)
+            cls = MagicMock(return_value=app)
+            setattr(importlib.import_module.return_value, appInfo.cls, cls)
+        self.buses = set()
+        for appInfo in APP_INFOS.values():
+            self.buses.update(appInfo.bus)
+        self.swift = Swift(APP_INFOS)
+
+    def test_init(self):
+        """Test if swift is initialized correctly."""
+        self.assertIsInstance(self.swift.mainWindow, QMainWindow)
+        self.assertIsInstance(self.swift.centralWidget, QLabel)
+        for name in APP_INFOS:
+            self.assertIn(name, self.swift._apps)
+        for bus in self.buses:
+            self.assertIn(bus, self.swift._subscribers)
+
+    def test_destroy_app(self):
+        """Test destroyApp()."""
+        name = list(APP_INFOS.keys())[0]
+        self.swift.destroyApp(name)
+        self.assertNotIn(name, self.swift._apps)
+
+    def test_broadcast(self):
+        """Test _broadcast()."""
+        busName = list(self.buses)[0]
+        self.swift._broadcast(busName, "test_msg")
+        for app in self.swift._subscribers[busName]:
+            app.received.emit.assert_called_once()
+
+
+class SwiftFunctionTest(unittest.TestCase):
+    """Unit test for functions in swift.py"""
+
+    def test_parse(self):
+        """Test parse()."""
+        self.assertEqual(parse(AppInfo, APP_JSONS["app1"]), APP_INFOS["app1"])
+        self.assertEqual(parse(AppInfo, APP_JSONS["app2_default"]), APP_INFOS["app2"])
+
+    def test_strinfo(self):
+        """Test strinfo()."""
+        self.assertEqual(strinfo(APP_INFOS["app1"]), APP_JSONS["app1"])
+        self.assertEqual(strinfo(APP_INFOS["app2"]), APP_JSONS["app2"])
+
+    def test_add_to_path(self):
+        """Test _add_to_path()."""
+        old_path = sys.path.copy()
+        with _add_to_path(os.path.dirname("test_dir")):
+            self.assertNotEqual(old_path, sys.path)
```

Comment:
Maybe checking if they are not equal is not enough to test this function.
In addition to that, please check if the test directory is _in_ `sys.path`.